### ISSUE TITLE
Training Segmentation and meets low speed issue regardless Large, Small or Nano model

### ISSUE THREAD

**Issue by LeMinhNgan** [roboflow/rf-detr#616]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

Hello all,

I am training Segmentation and meets low speed issue regardless Large, Small or Nano model: the training step is fast but very slow at validation step at the end of each epoch.
I tried  reducing size= 480, batch_size=1, use_ema=True, num_select=100 but still same issue. I am using RTX5090, 196GB ram, windows11

Any one having solution in this case?

### Environment

RF-DETR: 1.4.0
Windows11
Core ultra 5
192GB RAM
RTX 5090
Python 3.10

### Minimal Reproducible Example

```
import torch
from rfdetr RFDETRSegSmall

class Config():
    dataset_dir= r'D:\segmentation'                                      
    output_dir= r'D:\segmentation\outputs2'
    checkpoint= r'D:\segmentation\outputs2\checkpoint.pth'              

    net= RFDETRSegSmall                 
    resolution= 1008                    
    batch_size= 1                      
    lr= 1e-4
    checkpoint_interval= 5            
    epochs= 3                          
           
if __name__ == '__main__':
    device= 'cuda' if torch.cuda.is_available() else 'cpu'
    print('device=', device)
    cfg= Config()
    model= cfg.net(resolution= cfg.resolution, device= device)     
    model.train(dataset_dir= cfg.dataset_dir, epochs= cfg.epochs, batch_size= cfg.batch_size, grad_accum_steps=4, lr= cfg.lr, checkpoint_interval= cfg.checkpoint_interval, output_dir= cfg.output_dir, early_stopping= False, num_select=100, use_ema= True, resume= cfg.checkpoint, pretrain_weights= None)           
    
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by DKAvocad00** [comment]
i have same issue

**Comment by isaacrob** [comment]
It's slow due to conversion of masks to RLE via pycocotools. 

@Borda if you have any thoughts as to how to improve this experience, a clear place for gains! :)

**Comment by volnas10** [comment]
@isaacrob I understand converting hundreds of masks to RLE is probably the slowest part, but what about adding support for RLE annotations instead of polygons? `convert_coco_poly_to_mask()` converts the polygons to RLE anyway before creating masks so why not make the `ConvertCoco` class be able to load RLE directly? That would be one way to improve performance.

**Comment by isaacrob** [comment]
Don't see why not. Roboflow platform uses polygons which is why it's written that way but no reason to not support both

**Comment by dk-teknologisk-bgd** [comment]
I have the same issue. GPU utilization is very low during training

**Comment by ManuBN786** [comment]
I have same issue. It crashes at the first epoch too. takes forever to complete the first epoch and then crash !!

**Comment by Borda** [comment]
Hi @LeMinhNgan — apologies for the slow follow-up on this one. The slow validation is a known bottleneck from polygon→RLE mask conversion via pycocotools, not a regression. We're relabeling this as a performance enhancement. Native RLE annotation support (as @volnas10 suggested) is on our radar. #851 (lazy mask materialization, merged to develop, coming in the next release) should help with memory pressure during validation, though the core RLE overhead remains. Contributions welcome!

**Comment by Borda** [comment]
This is the same validation bottleneck tracked in #416 — `pycocotools` polygon→RLE conversion dominates validation time.

Workaround: set `eval_interval` (added in v1.6.0) to reduce eval frequency. Please follow #416 for updates on the lazy mask materialization fix.

Closing as duplicate — feel free to reopen!
